server2/Arcadia.Assistant/Arcadia.Assistant.EmailNotifications/EmailNotifications.cs

```diff
@@ -0,0 +1,90 @@
+namespace Arcadia.Assistant.EmailNotifications
+{
+    using System.Collections.Generic;
+    using System.Fabric;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Contracts;
+    using Contracts.Models;
+
+    using MailKit.Net.Smtp;
+    using MailKit.Security;
+
+    using Microsoft.Extensions.Logging;
+    using Microsoft.ServiceFabric.Services.Communication.Runtime;
+    using Microsoft.ServiceFabric.Services.Remoting.Runtime;
+    using Microsoft.ServiceFabric.Services.Runtime;
+
+    using MimeKit;
+
+    using Models;
+
+    /// <summary>
+    ///     An instance of this class is created for each service instance by the Service Fabric runtime.
+    /// </summary>
+    public class EmailNotifications : StatelessService, IEmailNotifications
+    {
+        private readonly ILogger logger;
+        private readonly SmtpSettings smtpSettings;
+
+        public EmailNotifications(
+            StatelessServiceContext context,
+            SmtpSettings smtpSettings,
+            ILogger<EmailNotifications> logger)
+            : base(context)
+        {
+            this.smtpSettings = smtpSettings;
+            this.logger = logger;
+        }
+
+        public async Task SendEmailNotification(
+            string[] recipients,
+            EmailNotificationContent notificationContent,
+            CancellationToken cancellationToken)
+        {
+            this.logger.LogDebug("Email notification message received");
```

Comment:
I'd wrap the whole method in try/catch with proper logging, so failing to send this notification doesn't stop all other notifications from being sent. Either that, or do it in Notifications service itself.

Reply:
Added exception handling blocks